File: backend/app/schemas/event.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class EventCreate(BaseModel):
    title: str
    event_type: str
    event_date: datetime
    location: str = ""
    notes: str = ""

class EventUpdate(BaseModel):
    title: Optional[str] = None
    event_type: Optional[str] = None
    event_date: Optional[datetime] = None
    location: Optional[str] = None
    notes: Optional[str] = None

class EventResponse(BaseModel):
    id: int
    title: str
    event_type: str
    event_date: datetime
    location: str
    notes: str
    outfit_generated: bool

    class Config:
        from_attributes = True

File: frontend/lib/models/event.dart
class AppEvent {
  final int id;
  final String title;
  final String eventType;
  final DateTime eventDate;
  final String location;
  final String notes;
  final bool outfitGenerated;

  AppEvent({
    required this.id,
    required this.title,
    required this.eventType,
    required this.eventDate,
    required this.location,
    required this.notes,
    required this.outfitGenerated,
  });

  factory AppEvent.fromJson(Map<String, dynamic> j) => AppEvent(
        id: j['id'],
        title: j['title'],
        eventType: j['event_type'],
        eventDate: DateTime.parse(j['event_date']),
        location: j['location'] ?? '',
        notes: j['notes'] ?? '',
        outfitGenerated: j['outfit_generated'] ?? false,
      );
}
